tests/test_skvbc_auto_view_change.py

```diff
@@ -0,0 +1,153 @@
+
+    def test_auto_vc_all_nodes_up_no_requests(self):
+        """
+        This test aims to validate automatic view change
+        in the absence of any client messages:
+        1) Start a full BFT network
+        2) Do nothing (wait for automatic view change to kick-in)
+        3) Check that view change has occurred (necessarily, automatic view change)
+        """
+        trio.run(self._test_auto_vc_all_nodes_up_no_requests)
+
+    async def _test_auto_vc_all_nodes_up_no_requests(self):
+        for bft_config in bft.interesting_configs():
+            config = bft.TestConfig(n=bft_config['n'],
+                                    f=bft_config['f'],
+                                    c=bft_config['c'],
+                                    num_clients=bft_config['num_clients'],
+                                    key_file_prefix=KEY_FILE_PREFIX,
+                                    start_replica_cmd=start_replica_cmd)
+            with bft.BftTestNetwork(config) as bft_network:
+                await bft_network.init()
+                bft_network.start_all_replicas()
+
+                initial_primary = 0
+
+                # do nothing - just wait for an automatic view change
+                await trio.sleep(seconds=10)
+
+                view = await bft_network.get_view_number(
+                    replica_id=random.choice(
+                        bft_network.all_replicas(without={initial_primary})),
+                    expected=lambda v: v > initial_primary
+                )
+                self.assertGreater(view, initial_primary,
+                                 "Make sure automatic view change has occurred.")
+
+    def test_auto_vc_when_primary_down(self):
+        """
+        This test aims to validate automatic view change
+        when the primary is down
+        1) Start a full BFT network
+        2) Stop the initial primary replica
+        3) Do nothing (wait for automatic view change to kick-in)
+        4) Check that view change has occurred (necessarily, automatic view change)
+        """
+        trio.run(self._test_auto_vc_when_primary_down)
+
+    async def _test_auto_vc_when_primary_down(self):
+        for bft_config in bft.interesting_configs():
+            config = bft.TestConfig(n=bft_config['n'],
+                                    f=bft_config['f'],
+                                    c=bft_config['c'],
+                                    num_clients=bft_config['num_clients'],
+                                    key_file_prefix=KEY_FILE_PREFIX,
+                                    start_replica_cmd=start_replica_cmd)
+            with bft.BftTestNetwork(config) as bft_network:
+                await bft_network.init()
+                bft_network.start_all_replicas()
+
+                initial_primary = 0
+                bft_network.stop_replica(initial_primary)
+
+                # do nothing - just wait for an automatic view change
+                await trio.sleep(seconds=10)
```

Comment:
same as above. Should probably be a 5 second sleep.

Reply:
see PR https://github.com/vmware/concord-bft/pull/320